frontend/src/components/podcast.tsx

```diff
@@ -0,0 +1,373 @@
+"use client"
+
+import { useEffect, useRef, useState, useTransition } from 'react'
+import { Play, Pause, SkipBack, SkipForward, Volume2, VolumeX, Trash2 } from 'lucide-react'
+import { PodcastUI } from '@/lib/podcast-ui'
+
+type Podcast = {
+  id: string
+  course_id: string
+  title: string
+  transcript: string
+  audio_path: string
+  storage_backend: string
+}
+
+function formatTime(seconds: number): string {
+  if (!isFinite(seconds)) return '0:00'
+  const m = Math.floor(seconds / 60)
+  const s = Math.floor(seconds % 60)
+  return `${m}:${s.toString().padStart(2, '0')}`
+}
+
+function PodcastPlayer({ podcastId, title, transcript, onDeleted }: { podcastId: string; title: string; transcript: string; onDeleted: () => void }) {
+  const audioRef = useRef<HTMLAudioElement | null>(null)
+  const [src, setSrc] = useState<string>(`/api/v1/podcasts/audio/${podcastId}`)
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [currentTime, setCurrentTime] = useState(0)
+  const [duration, setDuration] = useState(0)
+  const [volume, setVolume] = useState(1)
+  const [muted, setMuted] = useState(false)
+  const [resolving, setResolving] = useState(false)
+
+  // Resolve S3 presigned URL if backend returns JSON
+  async function resolveAudioSrc() {
+    try {
+      setResolving(true)
+      const res = await fetch(`/api/v1/podcasts/audio/${podcastId}`, {
```

Comment:
This is sufficient, but this is the right way to do it.

Create a Next.js API:
https://github.com/A3copilotprogram/plg_6/blob/main/frontend/src/app/api/v1/chat/%5BcourseId%5D/stream/route.ts

Consume the API:
https://github.com/A3copilotprogram/plg_6/blob/main/frontend/src/lib/chat.ts

Otherwise, you will run in the axios errors such as the ones I've pointed out.